embedding-sdk/authenticated/jwt/src/index.js

```diff
@@ -0,0 +1,69 @@
+import ChartsEmbedSDK from "@mongodb-js/charts-embed-dom";
+import "regenerator-runtime/runtime";
+import jwt from "jsonwebtoken";
+import config from "../config";
+
+document
+  .getElementById("loginButton")
+  .addEventListener("click", async () => await tryLogin());
+
+function getUser() {
+  return document.getElementById("username").value;
+}
+
+function getPass() {
+  return document.getElementById("password").value;
+}
+
+async function tryLogin() {
+  if (login(getUser(), getPass())) {
+    document.body.classList.toggle("logged-in", true);
+    await renderChart();
+  }
+}
+
+/*
+  login(username, password) is the function we are going to give to the
+  SDK to authenticate the user. It can be any function that returns a 
+  JWT token on success. The core of this function is the jwt.sign code,
+  which creates the token using the users username and our previously
+  defined secret called 'topsecret'.
+*/
+function login(username, password) {
+  // mock a check against the database
+  let mockedUsername = "admin";
+  let mockedPassword = "password";
+
+  if (username && password) {
+    if (username === mockedUsername && password === mockedPassword) {
+      let token = jwt.sign({ username: username }, config.secret, {
+        expiresIn: "24h" // expires in 24 hours
+      });
+      return token;
+    }
+  } else {
+    return false;
+  }
+}
+
```

Comment:
We can't have show a sample where the token is generated on the client; it's completely insecure. Can you please move this logic into a server-side API (Node is fine)?

Reply:
Sorry I guess that makes a lot of sense, went for too simple such that i made the example unrealistic. I'll remedy this